### ISSUE TITLE
Add Capitual - Multi-currency payment method; EUR, BRL, USD and GBP

### ISSUE THREAD

**Issue by pazza83** [bisq-network/growth#246]
## Capitual - Multi-currency

[Capitual Website](https://www.capitual.com/)
[Capitual Fees & Limits](https://exchange.capitual.com/)
[Capitual GitHub](https://github.com/capitual)
[Capitual Help Centre](https://help.capitual.com/en)
[Terms and conditions](https://my.capitual.com/legal#tos)
[Prohibited uses](https://static.capitual.net/legal/prohibited-businesses.pdf?view)
[Capitual Twitter](https://twitter.com/capitual)
[Jeton Blog](https://blog.jeton.com/)

Originally requested as a BRL payment method: https://github.com/bisq-network/growth/issues/215

### Background

Capitual appears to be a small Estonian based startup trying to combine fiat and crypto payments. They run an associated commerce based wallet called [CapPay](https://cappay.capitual.com/). 

They have good support at present for BRL and EUR. With free fiat deposits, withdrawals. They also offer users to send and receive fiat and Bitcoin between users for free. Fiat and other assets can also be swapped to and from crypto.

From the website:

> The world's first Hybrid Digital Bank. Where finances and technology merge into an inclusive future.

>Move fluidly through the most versatile platform on the market. Save, transfer and invest with a reliable and stable system, which gives a completely secure environment that allows the full management to be done seamlessly and privately.

>Whatever is your demand, Capitual wallets are ready to do the work! The most versatile and powerful wallets of the market. Create, manage and track all your wallets in real time. Send, receive, exchange and store fiat currencies, crypto-currencies and even precious metals.

>Where the best of cryptos and banking services meet. With Capitual, you can send and receive bank transfers, pay your bills directly from crypto-currencies, send invoices to anyone and keep track of your expenses for a simple but effective personal finances management.

### Region

Capitual say they operates in 48+ countries (US residents are excluded).

### Currencies

Multiple currencies are suported, these are:

- EUR
- BRL
- USD
- GBP

From what I can see only EUR and BRL are the ones that are able to be withdrawn / deposited to and from a fiat account via SEPA or TED (Express Wire Transfer).

USD and GBP can be added in the app by exchanging other currencies / cryptos.

Stable coins; BUSD, USDT, USDC and DAI can also be deposited and used to send / receive payments.

### Size of user base

Unknown

### Duration

Instant payment between users.

Crypto-currency deposit | Instant 
Crypto-currency withdraw | Instant
Fiat deposit | As soon as the funds reach our accounts; our detection of bank transfers are automated. Note that bank transfers may not be instant due to banking system's nature.
Fiat withdraw | From 1 minute to 48 working hours; bank transfers are mostly automated, but there might be cases when our system or banking partners identifies suspicious activities and it requires a manual approach from our team.

### Fees

It is free to transfer fiat / crypto to other users.

It is free to deposit, withdraw BRL and EUR.

It is free to deposit crypto / stablecoin funds.

Crypto withdrawal fees are:

- Bitcoin: BTC 0.0001
- Litecoin: LTC 0.01
- Dash: DSH 0.01
- Ethereum: ETH 0.01
- Tether: USDT 0.50
- Binance USD: BUSD 0.50
- USD Coin: USDC 0.50
- Dai Stablecoin: DAI 0.50

### Fraud risk
Low

### How payments are made

Transfers take place using CapCodes. CapCodes are like coupon codes you can generate at any time and use it to transfer funds

CapCodes are an electronic cheque system that makes it possible to store amounts of money, precious metals or cryupto-currencies in a short code, easy to be written down, that can be written in a piece of paper to make transactions even when you are not connected to the internet.

When you generate a CapCode, you get a voucher (coupon code) that you must take note (don’t worry: it’s always possible to recover a CapCode that you have lost!). Such code stores funds in any of the currencies supported by Capitual, and you can share it with someone else, allowing the payment in any of these currencies. It’s needed to have a Capitual account to redeem a CapCode.

After generating a CapCode, the amount gets “locked in”, and is already deducted from your balance. If you have given up or will no longer need to use the CapCode, you can redeem it yourself, onto your own account. It’s also possible to track, from your account, whether a CapCode has already been redeemed.

For while, CapCodes are only available when accessing your account using a computer.

### Payment amounts

Unconfirmed accounts have a fixed transfer limit of US$ 9,999.00. As this limit is reached, the account must be confirmed for continued usage.

**Comment by pazza83** [comment]
Capitual has been scored for inclusion to Bisq:


| Essential   | Desirable | Definite No’s |
| ------------------- | ------------------- | ------------------- |
| **Very low risk of chargeback**  | ~~No risk of chargeback~~  | ~~< very low risk of chargeback~~  |
| **Way to verify the sender in the received payment**  | ~~Way to verify the sender in the received payment and ability to enter a reference~~  | ~~No way to verify the sender in the received payment~~  |
| ~~Trade time less than one week~~ | **Instant payment**  | ~~Trade time more than one week~~  |
| ~~Singular Fiat currency~~  | **Multi-currency**  | ~~Not a payment method for fiat currency~~  |
| ~~Significant user base~~  |  ~~Large user base~~ | **No significant user base**  |
| **High usability**  | ~~High usability and great user experience~~  | ~~< high amount usability~~  |
| ~~No KYC required for sending and receiving payments~~ | **No KYC required for sending and receiving payments, allows users to trade with upmost privacy. Minimal identifying information as possible (no names, email, phone etc required)** | ~~Some KYC required (proof of address, ID, selfie) for sending and receiving payments~~  |
| **Low risk of scam attempts**  | ~~Very low risk of scam attempts~~  | ~~< low risk of scam attempts~~ |
| ~~Traders can provide evidence of payment / receipt~~  | **Traders can provide evidence of payment / receipt and Verification of payment can be made using PageSigner or similar** | ~~Traders will be unable to provide evidence of payment / receipt~~  |
| ~~Minimum limit at least equal to at least account limits protocols~~  | **No minimum limits**  | ~~Minimum limit not able to achieve account limits protocols~~  |
| **Maximum limits equal to at least 0.01 BTC**  | ~~Large payment limits up to 2 BTC~~ | ~~Maximum limit is less than 0.01 BTC~~  |
| ~~Likely to increase liquidity~~  | **Likely to increase liquidity and open markets for different countries and currencies**  | ~~Likely to decrease liquidity~~  |
| **Low risk of mediation**  | ~~Very low risk of mediation~~  | ~~< low risk of mediation~~  |
| **Low risk for traders from government agencies**  | ~~No risk for traders from government agencies~~ | ~~< low risk for traders from government agencies~~  |
| ~~Fees should not be a barrier to trading~~  | **No fees for transactions**  | ~~Fees will be a barrier to trading~~ |
| ~~Only minor changes needed to trade protocol~~  | **No changes needed to trade protocol**  | ~~> Minor changes needed to trade protocol~~  |

Score: 21/32
Percentage: 65.63%

**Comment by pazza83** [comment]
### Capitual - Proposed National Currency Payment Method 

I am proposing that Capitual be added as a national currency payment method.

@refund-agent2, @huey735, @leo816 please let me know if you have any questions as to how Capitual will be mediated / arbitrated. If you have any objection please let me know by 31 May 2021.

---

Date proposal being made: 02.05.21
Date feedback to be provided by: 31.05.21

Name of payment method: Capitual 

Required fields (eg)

- Need to confirm

Currencies accepted

- [ ] EUR
- [ ] BRL
- [ ] USD
- [ ] GBP


- Maximum trade duration: 24 hours

Signing required: No

Bisq Limits: eg 0.25 BTC

Trade window: 24 hours

Payment method limits: not aware of any

Any relevant information for users: None

Screenshot of expected GUI: TBC

**Comment by BtcContributor** [comment]
@pazza83 I started working on this.

Two questions:
1. Are there any specific limitations / constraints for CAP account number? So that I can implement proper checks and validators
2. Not sure which is the correct _Per-trade limit_ to set (See https://docs.bisq.network/payment-methods#supported-fiat-payment-methods). 

Thanks!

**Comment by pazza83** [comment]
Hi @BtcContributor 

0.25 BTC is used for low risk payment methods.

Generally fiat payment methods to not go above 0.25 BTC

**Comment by BtcContributor** [comment]
Hi @pazza83, implementation is almost completed.

Could you please tell me which are the exact words you want as Info when adding this new payment account?

For example, when you try to add a Revolut account you see this:

<img width="528" alt="info" src="https://user-images.githubusercontent.com/79100296/118294512-13ebf380-b4db-11eb-8d2d-46bfb0b0f0b4.png">

You can find other examples here and near that: https://github.com/bisq-network/bisq/blob/master/core/src/main/resources/i18n/displayStrings.properties#L3458

Maybe @m52go could help you. 

This info should be provided for all other new payments method to add.

Thanks!

**Comment by pazza83** [comment]
Hi @BtcContributor many thanks I have taken a look.

I think Capitual can be done with using the default messages. Seems that is what is used for most trades?


`portfolio.pending.step1.info=Deposit transaction has been published.\n{0} need to wait for at least one blockchain confirmation before starting the payment.`

`portfolio.pending.step2_buyer.pay=Please pay {0} via the specified payment method to the BTC seller. You''ll find the seller's account details on the next screen.\n\n`

`portfolio.pending.step2_seller.waitPayment.headline=Wait for payment`

`portfolio.pending.step2_seller.waitPayment.msg=The deposit transaction has at least one blockchain confirmation.\nYou need to wait until the BTC buyer starts the {0} payment.`

With regards the information tag I am not sure when this appears in the process. I am assuming when something has changed from one Bisq version to another?

@m52go please can you let me know if you have a template you would like me to use for the text to add the payment methods. I have not completed any in app text before. Also it would be good to know what the text defaults too if nothing is entered.

Many thanks

**Comment by BtcContributor** [comment]
Hi @pazza83, you missed step3 also, could you please take a look?

As for the info message you can find that when you a add a new payment account. Try to add a Revolut account and you will see where appears the screenshot in my previous post.

Once I have the final text, I can release the final code for Capital.

**Comment by m52go** [comment]
I'm working with pazza83 to figure out this text -- my fault for missing this exchange earlier. 

There's no template yet, but I think it would be a good idea to establish some kind of general spec to make implementing new payment methods straightforward for any developer.

**Comment by pazza83** [comment]
Hi @m52go 

I am scratching my head a bit with this task!!

Here is the reference file I am using: 
https://github.com/bisq-network/bisq/blob/master/core/src/main/resources/i18n/displayStrings.properties

I think the custom strings for a 'NEW-PAYMENT-METHOD' that need to be defined by the payment maintainer are as follows:

`portfolio.pending.step2_buyer.NEW-PAYMENT-METHOD=Please pay {0} to the BTC seller by using NEW-PAYMENT-METHOD (include any things to for buyer to be aware of)
`

`portfolio.pending.step3_seller.NEW-PAYMENT-METHOD=The buyer has to sent you.... please check.... Only confirm receipt after you have received the funds. (include any things to for the seller to be aware of)
`

`payment.NEW-PAYMENT-METHOD.info=Include any special instructions for the payment method that both buyer and seller might need to be aware of.`

Additional custom fields for the fields themselves but I think these will all be self explanatory (based on field names as shown in graphic examples) and can be added by the devs? 

Let me know your thoughts.

**Comment by pazza83** [comment]
Wiki completed: https://bisq.wiki/Capitual
